components/raftstore/src/store/peer.rs

```diff
@@ -1631,6 +1634,10 @@ where
                 }
                 self.post_pending_read_index_on_replica(ctx);
                 self.read_progress.update_applied(self.last_applying_idx);
+                if !self.pending_messages.is_empty() {
```

Comment:
There seems to be a race condition between canceling snapshot and generating ready. If a snapshot is canceled, `check_applying_snap` will return Idle.

Reply:
Commented on `CheckApplyingSnapStatus::Idle`.